boxes/boxes/builder.py

```diff
@@ -0,0 +1,163 @@
+import pathlib
+import logging
+import tempfile
+import shutil
+import os
+import tarfile
+from typing import List, Optional
+
+import boxes.utils.cmd
+import boxes.boxes
+
+
+class Builder:
+    def __init__(
+        self,
+        destination: pathlib.Path,
+        chroot_path: Optional[pathlib.Path],
+        release: str,
+        extract: bool
+    ):
+        self._destination = destination
+        self._extract = extract
+        self._logger = logging.getLogger('builder')
+        self._cmd = boxes.utils.cmd.CommandRunner(self._logger)
+        self._release = release
+
+        if chroot_path is None:
+            self._workdir = pathlib.Path(
+                tempfile.mkdtemp(prefix='sio2jail-box')
+            )
+            self._root = self._workdir / 'root'
+            self._root.mkdir()
+            self._do_prepare_chroot = True
+        else:
+            self._workdir = None
+            self._root = chroot_path.resolve()
+            if self._root.exists():
+                self._do_prepare_chroot = False
+            else:
+                self._root.mkdir()
+                self._do_prepare_chroot = True
+
+    def __del__(self) -> None:
+        if not self._workdir:
+            return
+        try:
+            with self._cmd.sudo():
+                self._cmd.run('chmod', '-R', 'a+rwX', str(self._workdir))
+            self._logger.info("Deleting workdir %s", self._workdir)
+            shutil.rmtree(str(self._workdir))
+        except Exception:
+            self._logger.exception("Failed to delete workdir %s", self._workdir)
+
+    def run(self, boxes_names: List[str]):
+        if self._do_prepare_chroot:
+            self._prepare_chroot()
+        for box in (boxes_names or boxes.boxes.DEFAULT_BOXES):
```

Comment:
Ja bym to DEFAULT_BOXES dał poziom wyzej, tzn. w tym co wywołuje run(). Ale może zostać.

Reply:
Chciałem, żeby było tam, gdzie definicje, ale trochę w/e

Reply:
W sensie, zdefiniowane może być tam gdzie jest, ale chodzi mi o to gdzie jest użyte.

Reply:
A spoko, w sumie nie pamiętam dlaczego tak do tego wróciłem

Reply:
fix'd